### DISCUSSION TITLE
useContractWrite with usePrepareContractWrite gives type error for config

### DISCUSSION THREAD

**Discussion by technophile-04** [wevm/wagmi#1286] [Q&A] [answered]
### current code : 
```ts
  const { config, error: inputError } = usePrepareContractWrite({
    address: contractAddress,
    functionName: functionFragment.name,
    abi: [functionFragment],
    args: keys.map(key => form[key]),
    overrides: {
      value: ethers.BigNumber.from(txValue),
    },
  });

  const { data: result, isLoading, writeAsync } = useContractWrite(config);
```

### error : 
```js
Argument of type 'PrepareWriteContractResult<[FunctionFragment], string>' is not assignable to parameter of type 'UseContractWriteConfig<readonly unknown[], string> | undefined'.
```

### Full error : 
![Screenshot 2022-11-11 at 11 28 01 PM](https://user-images.githubusercontent.com/80153681/201400686-2abce442-b7d1-48d8-b554-9734a2098a78.jpg)

**Comment by tmm** [comment]
Can you post a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? It's difficult to look into this without full information.

**Reply by technophile-04** [reply]
Thanks, @tmm for the quick reply, 
For minimal reproducible example, i just created a repo similar to [rainbow-next-mintNFT](https://github.com/rainbow-me/rainbowkit/tree/main/examples/with-next-mint-nft) example and just removed `as const` from [contract-abi.ts](https://github.com/rainbow-me/rainbowkit/blob/main/examples/with-next-mint-nft/contract-abi.ts)
repo for minimal reproducible example error : https://github.com/technophile-04/minimal-rainbow-next-mintNFT

if you clone the repo and checkout [pages/index.tsx](https://github.com/technophile-04/minimal-rainbow-next-mintNFT/blob/main/pages/index.tsx) you will see a similar error. 

I think its something to do with inferring of types as mentioned in docs [here](https://wagmi.sh/docs/typescript#type-inference). 
Is there a way out of this other than @ts-ignore? since `abi` for me is dynamic .

```js
const { config, error: inputError } = usePrepareContractWrite({
    address: contractAddress,
    functionName: functionFragment.name,
    abi: [functionFragment], // <--- dynamic
    args: keys.map(key => form[key]),
    overrides: {
      value: ethers.BigNumber.from(txValue),
    },
  });

  const { data: result, isLoading, writeAsync } = useContractWrite(config);
```

**Comment by qqqllppp** [comment, marked as answer]
I am getting a similar error when I `build`. This occured after upgrading to `wagmi@0.7.14`.

My temporary fix based on https://wagmi.sh/docs/typescript#type-inference:
```
const { config } = usePrepareContractWrite({ ... })
const { abi, ...config_ } = config;
const { ... } = useContractWrite({ ...config_, ...([abi] as const) })
```

**Reply by technophile-04** [reply]
Thanks so much, @junhuang-ho for this 🙏 🙌, it worked  !!

**Reply by qqqllppp** [reply]
but of course I expect that we'll be able to directly pass `config` directly into `useContractWrite` without needing to deconstruct it in the next wagmi patch @tmm ?

**Reply by tmm** [reply]
[`wagmi@0.8.3`](https://github.com/wagmi-dev/wagmi/releases/tag/wagmi%400.8.3) should take care of this.
